lib/rubocop/cop/rspec/undescriptive_literals_description.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Description should be descriptive.
+      #
+      # If example group or example contains only `execute string`, numbers
+      # and regular expressions, the description is not clear.
+      #
+      # @example
+      #   # bad
+      #   describe `time` do
+      #    # ...
+      #   end
+      #
+      #   # bad
+      #   context /when foo/ do
+      #     # ...
+      #   end
+      #
+      #   # bad
+      #   it 10000 do
+      #     # ...
+      #   end
+      #
+      #   # good
+      #   describe Foo do
+      #     # ...
+      #   end
+      #
+      #   # good
+      #   describe '#foo' do
+      #     # ...
+      #   end
+      #
+      #   # good
+      #   context "when #{foo} is bar" do
+      #     # ...
+      #   end
+      #
+      #   # good
+      #   it 'does something' do
+      #     # ...
+      #   end
+      #
+      class UndescriptiveLiteralsDescription < Base
+        MSG = 'Description should be descriptive.'
+
+        # @!method example_groups_or_example?(node)
+        def_node_matcher :example_groups_or_example?, <<~PATTERN
+          (block (send #rspec? {#ExampleGroups.all #Examples.all} $_) ...)
+        PATTERN
+
+        def on_block(node) # rubocop:disable InternalAffairs/NumblockHandler
+          example_groups_or_example?(node) do |arg|
+            add_offense(arg) if offense?(arg)
+          end
+        end
+
+        private
+
+        def offense?(node)
+          if node.send_type? && node.method?(:+)
```

Comment:
You’ve added a test for foo + time, but only when all? is false.

If your experience suggests that we need to dive as deep as handling +, I think that if any of parts is an xstr, the whole thing is non-deterministic.
Is sometimh like “‘time’ + ‘whoami’” real? Or just as mistyped quotes like “‘when it is’ + time”?

Again, no objections on merging.

Reply:
Indeed it is! If xstr is included when concatenated with `+`, it's an empty string, so it might as well be offense. In the case of int and regexp, there are cases of false positives, as in the case of the added tests, so I thought it would be a good idea to accept them uniformly. What do you think?

```
❯ irb
irb(main):001> `time` + `time`
=> ""
irb(main):002> `time` + 1
(irb):2:in `+': no implicit conversion of Integer into String (TypeError)
        from (irb):2:in `<main>'
        from <internal:kernel>:187:in `loop'
        from /ydah/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/gems/irb-1.11.2/exe/irb:9:in `<top (required)>'
        from /ydah/.rbenv/versions/3.3.0/bin/irb:25:in `load'
        from /ydah/.rbenv/versions/3.3.0/bin/irb:25:in `<main>'
irb(main):003> `time` + 'foo'
=> "foo"
irb(main):004> 'when' + `time`
=> "when"
```

Reply:
Or it might be better to always allow for concatenation by `+`. Because it seems difficult to understand the violation in case of violation, and it seems to be outside the scope of this cop.

```ruby
  it 'registers an offense when using `describe` with `time`+"foo"' do
    expect_offense(<<~RUBY)
      describe `time` + 'foo' do
               ^^^^^^^^^^^^^^ Description should be descriptive.
      end
    RUBY
  end

  it 'registers an offense when using `describe` with "foo"+`time`' do
    expect_offense(<<~RUBY)
      describe 'foo' + `time` do
               ^^^^^^^^^^^^^^ Description should be descriptive.
      end
    RUBY
  end

  it 'registers an offense when using `describe` with "foo+(`time`+"bar")' do
    expect_offense(<<~RUBY)
      describe 'foo' + (`time` + 'bar') do
               ^^^^^^^^^^^^^^^^^^^^^^^^ Description should be descriptive.
      end
    RUBY
  end
```